src/cursor.c

```diff
@@ -98,11 +98,11 @@ cursor_get_from_ssd(enum ssd_part_type view_area)
 static struct wlr_surface *
 get_toplevel(struct wlr_surface *surface)
 {
-	while (surface && wlr_surface_is_xdg_surface(surface)) {
+	while (surface) {
```

Comment:
I don't remember the details but I think this check was important, maybe it was regarding layershell surfaces?

Reply:
wlroots removed the xxx_is_yyy() functions and now we just have to check for yyy_try_from_xxx() returning NULL. There's a check for that inside the loop below, which should be equivalent to the old logic.

Reply:
I've just read it again and I think you are right. The logic seems the same after the change. Sorry for the noise.